pandas/core/series.py

```diff
@@ -1429,8 +1429,55 @@ def mode(self):
         # TODO: Add option for bins like value_counts()
         return algorithms.mode(self)
 
-    @Appender(base._shared_docs['unique'] % _shared_doc_kwargs)
     def unique(self):
+        """
```

Comment:
good point but they have some differences:
* pd.unique takes param and Series.unique doesn't take.
* pd.unique handles 1d array-like objects including Index and Series.unique applies on self.
* pd.unique examples contain more and Series.unique only series examples

do we have pattern somewhere in regards to conditionally show docstring lines?

Reply:
https://github.com/pandas-dev/pandas/pull/20361/files is doing something similar for factorize. It's somewhat complex, since we have `pd.unique`, `Series/Index.unique`, and `Categorical.unique`. I'd be OK with improving the docstring here, and merging theme later.

Reply:
agree, this PR is mainly about improve docstring. and should be another PR synthesising docs of all unique methods.

Reply:
that's fine too